src/Rules/Operators/InvalidBinaryOperationRule.php

```diff
@@ -71,22 +72,28 @@ public function processNode(Node $node, Scope $scope): array
 				$callback = static fn (Type $type): bool => !$type->toNumber() instanceof ErrorType;
 			}
 
-			$leftType = $this->ruleLevelHelper->findTypeToCheck(
-				$scope,
-				$left,
-				'',
-				$callback,
-			)->getType();
+			$leftType = $scope->getType($left);
+			if (!$leftType instanceof MixedType) {
```

Comment:
RuleLevelHelper contains an early return for MixedType, therefore we don't call into it on Mixed

Reply:
I don't understand this change.

Reply:
see https://github.com/phpstan/phpstan-src/pull/1579#discussion_r955186491